base/poco/Net/src/HTTPSession.cpp

```diff
@@ -70,7 +70,7 @@ HTTPSession::~HTTPSession()
 {
 	try
 	{
-		if (_pBuffer) HTTPBufferAllocator::deallocate(_pBuffer, HTTPBufferAllocator::BUFFER_SIZE);
+		if (_pBuffer) delete[] _pBuffer;
```

Comment:
Maybe change _pBuffer from char* to std::vector<char> or std::unique_ptr<char[]>  so that manual delete is not needed?
Also try-catch around delete is not not needed after you got rid of the pool.